codecs-parent/codecs-framework/src/test/java/org/interledger/encoding/asn/serializers/oer/UnsignedLongOerSerializerTest.java

```diff
@@ -0,0 +1,152 @@
+            // 14
+            {UnsignedLong.valueOf("16777217"), Longs.toByteArray(16777217L)},
+            // 15
+            {UnsignedLong.valueOf("4294967294"), Longs.toByteArray(4294967294L)},
+            // 16
+            {UnsignedLong.valueOf("4294967295"), Longs.toByteArray(4294967295L)},
+
+            // Five Bytes (40 bits)
+            // 17
+            {UnsignedLong.valueOf("4294967296"), Longs.toByteArray(4294967296L)},
+            // 18
+            {UnsignedLong.valueOf("4294967297"), Longs.toByteArray(4294967297L)},
+            // 19
+            {UnsignedLong.valueOf("1099511627774"), Longs.toByteArray(1099511627774L)},
+            // 20
+            {UnsignedLong.valueOf("1099511627775"), Longs.toByteArray(1099511627775L)},
+
+            // Six Bytes (48 bits)
+            // 21
+            {UnsignedLong.valueOf("1099511627776"), Longs.toByteArray(1099511627776L)},
+            // 22
+            {UnsignedLong.valueOf("1099511627777"), Longs.toByteArray(1099511627777L)},
+            // 23
+            {UnsignedLong.valueOf("281474976710654"), Longs.toByteArray(281474976710654L)},
+            // 24
+            {UnsignedLong.valueOf("281474976710655"), Longs.toByteArray(281474976710655L)},
+
+            // Seven Bytes (56 bits)
+            // 25
+            {UnsignedLong.valueOf("281474976710656"), Longs.toByteArray(281474976710656L)},
+            // 26
+            {UnsignedLong.valueOf("281474976710657"), Longs.toByteArray(281474976710657L)},
+            // 27
+            {UnsignedLong.valueOf("72057594037927934"), Longs.toByteArray(72057594037927934L)},
+            // 28 (max 7-bit long value)
+            {UnsignedLong.valueOf("72057594037927935"), Longs.toByteArray(72057594037927935L)},
+
+            // 29
+            {UnsignedLong.valueOf("9223372036854775805"), Longs.toByteArray(Long.MAX_VALUE - 2L)},
+            // 30
+            {UnsignedLong.valueOf("9223372036854775806"), Longs.toByteArray(Long.MAX_VALUE - 1L)},
+            // 31
+            {UnsignedLong.valueOf("9223372036854775807"), Longs.toByteArray(Long.MAX_VALUE)},
+
+            // 32 Eight bytes, beyond Long.MAX_VALUE
+            {UnsignedLong.valueOf("9223372036854775808"),
+                new byte[] {(byte) 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+
+            // 33
+            {UnsignedLong.valueOf("9223372036854775809"),
+                new byte[] {(byte) 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}},
+            // 33
+            {UnsignedLong.valueOf("18446744073709551614"),
+                new byte[] {(byte) 0xFF, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF,
+                    (byte) 0xFF, (byte) 0xFF, (byte) 0xFE}},
+            // 34
+            {UnsignedLong.valueOf("18446744073709551615"),
+                new byte[] {(byte) 0xFF, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF, (byte) 0xFF,
+                    (byte) 0xFF, (byte) 0xFF, (byte) 0xFF}}
+        }
```

Comment:
It'd be nice to have an explanation or a link pointing to the documentation or the standard that defines this behavior. Other languages that might view quilt as a baseline and implement their equivalents might need to know exactly how to handle these overflows which are greater than the standard supported bounds.

Reply:
While these _are_ overflows for a typical Java `long` number (which is signed), these aren't overflows for `UnsignedLong`, which can hold up to `2^64 - 1`